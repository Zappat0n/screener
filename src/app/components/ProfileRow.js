import PropTypes from 'prop-types';

const ProfileRow = ({
  title1, data1, title2, data2,
}) => (
  <div className="row">
    <div className="cell">
      <div>{title1}</div>
      <div>{data1}</div>
    </div>
    <div className="cell">
      <div>{title2}</div>
      <div>{data2}</div>
    </div>
  </div>
);

ProfileRow.defaultProps = {
  data1: '',
  data2: '',
};

ProfileRow.propTypes = {
  title1: PropTypes.string.isRequired,
  data1: PropTypes.string,
  title2: PropTypes.string.isRequired,
  data2: PropTypes.string,
};

export default ProfileRow;
