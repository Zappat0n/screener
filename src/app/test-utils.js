import React from 'react';
import { createStore, combineReducers } from 'redux';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import indexReducer from './reducers/indexSlice';
import stockReducer from './reducers/stockSlice';
import profileReducer from './reducers/profileSlice';
import filterReducer from './reducers/filterSlice';

const rootReducer = combineReducers({
  indexes: indexReducer, stocks: stockReducer, profile: profileReducer, filter: filterReducer,
});

const initialState = {
  stocks: {
    AAPL: {
      price: 100,
      companyName: 'Apple Inc',
      changes: 1,
      sector: 'Technology',
    },
  },
  profile: {
    data: {
      image: '',
      companyName: 'Microsoft',
      price: 100,
      changes: 1,
      mktCap: 1000000,
      range: '0-100',
      country: 'US',
    },
  },
};

const renderComponent = (component) => render(
  <Provider store={createStore(rootReducer, initialState)}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export * from '@testing-library/react';
export { renderComponent };
