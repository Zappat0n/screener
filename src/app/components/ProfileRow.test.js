import React from 'react';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import ProfileRow from './ProfileRow';
import { renderComponent } from '../test-utils';

renderComponent(
  <ProfileRow title1="Name" data1="Me" title2="Sector" data2="Mine" />,
);

test('Renders Index correctly', () => {
  expect(screen.getByText('Name')).toBeInTheDocument();
  expect(screen.getByText('Me')).toBeInTheDocument();
  expect(screen.getByText('Sector')).toBeInTheDocument();
  expect(screen.getByText('Mine')).toBeInTheDocument();
});
