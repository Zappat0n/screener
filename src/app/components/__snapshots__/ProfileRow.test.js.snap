// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileRow Match SnapShot 1`] = `
<div
  className="row"
>
  <div
    className="cell"
  >
    <div>
      Name
    </div>
    <div>
      Me
    </div>
  </div>
  <div
    className="cell"
  >
    <div>
      Sector
    </div>
    <div>
      Mine
    </div>
  </div>
</div>
`;
